### PR TITLE
Add some Quality of Life improvements

### LINKED ISSUE #13: split dev and non-dev requirements
https://github.com/developmentseed/remote-workstation/blob/c985c675ce201d6c5492854a7ffacba2d2198a4f/Pipfile#L6-L18

### PR DESCRIPTION
**Pull request by ciaransweet**
### TL;DR:

This PR adds some Quality of Life improvements to the repository, namely pinning versions and making overall interaction with the project a bit easier

Closes #13 #12

### Dependencies

* Pinned dependency versions in both `Pipfile` and `package-lock.json`
* Add `Makefile` commands to abstract installation of 'prod' vs 'dev' dependencies

### CDK

* Moved all CDK related code into `cdk/` so that we're explicitly keeping it in one location and not the root and another directory

### General

* Add a more verbose and wider-ranging `.gitignore`
* Align `Makefile` to match the more recent style I've used in `pangeo-forge-aws-bakery`
* Add `make ssh-to-instance` to abstract out ssh'ing logic
* Update `README.md`:
    * Add build status badge
    * Update pre-requisites to match `pangeo-forge-aws-bakery`s style
    * Update `Makefile` section
    * Add some links in the `.env` table
    * Extract the example `.env` file out to `example.env`

### Github

* Add basic Github Actions workflow
* Add PR template

## How you can test it

No real functional changes as such, a CI build should happen which should pass on `linting`.

Otherwise you can re-run through the instructions and see that the deployment still functions. 

### Note:

I do intend on adding some unit tests for the `utils` code eventually.

### CODE REVIEW

**Review by vincentsarago — DISMISSED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -69,7 +79,7 @@ $ make deploy
 ## 5. SSH to your instance
 
 ```bash
-$ ssh remote-workstation-<value of IDENTIFIER>
+$ make ssh-to-instance
```
Comment (wildintellect) [inline README.md:82]: Is this so the user doesn't have to remember where the instance is, but rather it's written in a local config? or is it still in the system-wide ssh config?

**Comment by wildintellect** [comment]
Makes me realize at some point we might need to package this so it's a utility that can be installed and/or run on a per repo/project basis.

I'm still a little confused with this change, if someone deploys, ssh, starts some code, disconnects, and then wants to come back later to check on the instance, how they remember which instance to ssh to?

**Review by wildintellect — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ciaransweet) [reply README.md:82]: We write to either the local config or where they specify `SSH_CONFIG_LOCATION`

`make ssh-to-instance` now means that you don't need to provide anything else.

If `SSH_CONFIG_LOCATION` is populated, it'l use that and `IDENTIFIER` if that's populated, else we default to the locally created one and `dev`.

You shouldn't need nor care where the instance is

**Comment by ciaransweet** [comment]
> Makes me realize at some point we might need to package this so it's a utility that can be installed and/or run on a per repo/project basis.
> 
> I'm still a little confused with this change, if someone deploys, ssh, starts some code, disconnects, and then wants to come back later to check on the instance, how they remember which instance to ssh to?

@wildintellect not sure what you're implying with this? What would be packaged as a utility?

**Comment by ciaransweet** [comment]
Also @wildintellect if something doesn't make sense, you're welcome to reject the PR haha!
